Tests/TuistKitTests/Services/DocServiceTests.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistDoc
+import TuistDocTesting
+import TuistSupport
+import XCTest
+
+@testable import TuistCoreTesting
+@testable import TuistKit
+@testable import TuistSupportTesting
+
+final class TuistDocServiceTests: TuistUnitTestCase {
+    var subject: DocService!
+
+    var projectGenerator: MockProjectGenerator!
+    var swiftDocController: MockSwiftDocController!
+    var opener: MockOpener!
+    var swiftDocServer: MockSwiftDocServer!
+
+    override func setUp() {
+        super.setUp()
+
+        projectGenerator = MockProjectGenerator()
+        swiftDocController = MockSwiftDocController()
+        opener = MockOpener()
+        swiftDocServer = MockSwiftDocServer()
+        swiftDocServer.stubBaseURL = "http://tuist.io"
+
+        fileHandler = MockFileHandler(temporaryDirectory: { try self.temporaryPath() })
+
+        subject = DocService(projectGenerator: projectGenerator,
+                             swiftDocController: swiftDocController,
+                             swiftDocServer: swiftDocServer,
+                             fileHandler: fileHandler)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+
+        projectGenerator = nil
+        swiftDocController = nil
+        opener = nil
+        swiftDocServer = nil
+        subject = nil
+    }
+
+    func test_doc_fail_missing_target() {
+        let path = AbsolutePath("/.")
+        XCTAssertThrowsError(try subject.run(project: path, target: "CustomTarget", serve: false, port: 4040))
+    }
+
+    func test_doc_fail_missing_file() {
```

Comment:
I think you can also add `// Given // When // Then` convention that is used throughout most of our tests